### ISSUE TITLE
使用python api gpu 占用为0问题

### ISSUE THREAD

**Issue by langwenjing** [PaddlePaddle/Paddle#6378]
使用python api训练模型，指定了paddle.init(use_gpu=True, trainer_count=4)，并且设置了CUDA_VISIBLE_DEVICES，启动发现gpu占用为0，请问还需要配置什么参数吗？
并且程序运行一段时间出现中断：
*** Aborted at 1512634078 (unix time) try "date -d @1512634078" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGFPE (@0x7f19e84f5e39) received by PID 14737 (TID 0x7f1a38005700) from PID 18446744073312099897; stack trace: ***
    @     0x7f1a37bdc160 (unknown)
    @     0x7f19e84f5e39 paddle::AucEvaluator::evalImp()
    @     0x7f19e84f6805 paddle::Evaluator::eval()
    @     0x7f19e84c2048 paddle::CombinedEvaluator::eval()
    @     0x7f19e84c73f2 paddle::MultiGradientMachine::eval()
    @     0x7f19e8377daa _wrap_GradientMachine_eval
    @           0x4b4cb9 PyEval_EvalFrameEx
    @           0x4b6b28 PyEval_EvalCodeEx
    @           0x4b5d10 PyEval_EvalFrameEx
    @           0x4b6b28 PyEval_EvalCodeEx
    @           0x4b5d10 PyEval_EvalFrameEx
    @           0x4b6b28 PyEval_EvalCodeEx
    @           0x4b5d10 PyEval_EvalFrameEx
    @           0x4b5fb8 PyEval_EvalFrameEx
    @           0x4b6b28 PyEval_EvalCodeEx
    @           0x4b6c52 PyEval_EvalCode
    @           0x4e1c7d PyRun_FileExFlags
    @           0x4e3501 PyRun_SimpleFileExFlags
    @           0x4159dd Py_Main
    @     0x7f1a37136bd5 __libc_start_main
    @           0x414b71 (unknown)
    @                0x0 (unknown)

请帮忙给出gpu训练方法，谢谢~

**Comment by peterzhang2029** [comment]
你好 请贴一下你的具体配置 以便排查问题

**Comment by langwenjing** [comment]
paddle.init(use_gpu=True, trainer_count=4)
    dict_dim = len(word_dict)
    # network config
    cost, prob, label = convolution_net(dict_dim, class_num)

    # create parameters
    parameters = paddle.parameters.create(cost)
 
    parameters.set('emb_q1q2', load_parameter(train_data_emb_path, dict_dim, 500))
    parameters.set('emb_q3', load_parameter(train_data_emb_path, dict_dim, 500))
    # create optimizer
    adam_optimizer = paddle.optimizer.Adam(
        learning_rate=1e-3,
        regularization=paddle.optimizer.L2Regularization(rate=1e-3),
        model_average=paddle.optimizer.ModelAverage(average_window=0.5))

    # create trainer
    trainer = paddle.trainer.SGD(
        cost=cost,
        extra_layers=paddle.evaluator.auc(input=prob, label=label),
        parameters=parameters,
        update_equation=adam_optimizer)

**Comment by peterzhang2029** [comment]
麻烦把auc去掉试试能不能跑，请确认gpu的数量要小于等于4 并且可以先测试cpu下是否可以执行。

**Comment by langwenjing** [comment]
auc已经删掉。cpu下可以正常执行。机器是16个gpu，配置export CUDA_VISIBLE_DEVICES=6,7，使用两个。paddle.init(use_gpu=True, trainer_count=4)，启动直接报错：
I1207 18:50:02.662008  2940 MultiGradientMachine.cpp:99] numLogicalDevices=1 numThreads=4 numDevices=2
F1207 18:50:02.662045  2940 MultiGradientMachine.cpp:104] trainer_count * num_devices_in_model (4*1)=4 exceeds number of GPU devices(2)

**Comment by peterzhang2029** [comment]
@langwenjing 这个报错是因为你设置的` export CUDA_VISIBLE_DEVICES=6,7`只有两个GPU可用，而你的paddle里设置的trainer_count=4需要4卡，所以报错了。
